src/RoboSaverVirtualModule.sol

```diff
@@ -231,18 +233,27 @@ contract RoboSaverVirtualModule {
     function checker() external view returns (bool adjustPoolNeeded, bytes memory execPayload) {
         if (_isExternalTxQueued()) return (false, bytes("External transaction in queue, wait for it to be executed"));
 
-        /// @notice checks if there is a transaction queued up in the delay module by the virtual module itself
+        /// @dev check if there is a transaction queued up in the delay module by the virtual module itself
         if (txQueueData.nonce != 0) {
             return (true, abi.encodeWithSelector(this.adjustPool.selector, PoolAction.EXEC_QUEUE_POOL_ACTION, 0));
         }
 
         uint256 balance = EURE.balanceOf(CARD);
+        if (balance == 0) revert ZeroBalance();
+
         (, uint128 dailyAllowance,,,) = rolesModule.allowances(SET_ALLOWANCE_KEY);
 
         if (balance < dailyAllowance) {
             /// @notice there is a deficit; we need to withdraw from the pool
+            uint256 bptBalance = BPT_STEUR_EURE.balanceOf(CARD);
+            if (bptBalance == 0) revert ZeroBalance();
```

Comment:
actually in the `checker` should not revert per se. rather return the right msg format

i.e:`false, bytes("No BPT balancer")`

Reply:
b5ccb45